static/app/utils/projects.tsx

```diff
@@ -149,6 +149,37 @@ class Projects extends React.Component<Props, State> {
     }
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const {projects} = this.props;
+
+    if (projects !== prevProps.projects) {
+      this.updateProjectsFromStore();
+    }
+  }
+
+  /**
+   * Function to update projects when the store emits updates
+   */
+  updateProjectsFromStore = () => {
+    const {allProjects, projects, slugs} = this.props;
+
+    if (allProjects) {
+      this.setState({
+        fetchedProjects: projects,
+      });
+      return;
+    }
+
+    if (slugs && !!slugs.length) {
+      // Extract the requested projects from the store based on props.slugs
+      const projectsMap = this.getProjectsMap(projects);
+      const projectsFromStore = slugs.map(slug => projectsMap.get(slug)).filter(defined);
+      this.setState({
```

Comment:
this is the most minor style nit, but I tend to like to save a line or two on things like this and just keep it one line. I think it's still pretty readable

`this.setState({projectsFromStore})`